Modules/Challenges/test/Challenges.Application.Tests/Tests/Identities/IdentityDeleterTests.cs

```diff
@@ -1,25 +1,44 @@
-﻿using Backbone.Modules.Challenges.Application.Challenges.Commands.DeleteChallengesOfIdentity;
+﻿using Backbone.BuildingBlocks.Application.Identities;
+using Backbone.Modules.Challenges.Application.Challenges.Commands.DeleteChallengesOfIdentity;
 using Backbone.Modules.Challenges.Application.Identities;
-using Backbone.UnitTestTools.Data;
+using Backbone.UnitTestTools.BaseClasses;
 using FakeItEasy;
 using MediatR;
 using Xunit;
+using static Backbone.UnitTestTools.Data.TestDataGenerator;
 
 namespace Backbone.Modules.Challenges.Application.Tests.Tests.Identities;
-public class IdentityDeleterTests
+public class IdentityDeleterTests : AbstractTestsBase
 {
     [Fact]
     public async Task Deleter_calls_correct_command()
     {
         // Arrange
         var mockMediator = A.Fake<IMediator>();
-        var identityAddress = TestDataGenerator.CreateRandomIdentityAddress();
         var deleter = new IdentityDeleter(mockMediator);
+        var identityAddress = CreateRandomIdentityAddress();
+        var dummyIDeletionProcessLogger = A.Fake<IDeletionProcessLogger>();
 
         // Act
-        await deleter.Delete(identityAddress);
+        await deleter.Delete(identityAddress, dummyIDeletionProcessLogger);
 
         // Assert
         A.CallTo(() => mockMediator.Send(A<DeleteChallengesOfIdentityCommand>.That.Matches(command => command.IdentityAddress == identityAddress), A<CancellationToken>._)).MustHaveHappened();
     }
+
+    [Fact]
+    public async Task Deleter_correctly_creates_audit_log()
+    {
+        // Arrange
+        var dummyMediator = A.Fake<IMediator>();
+        var deleter = new IdentityDeleter(dummyMediator);
+        var identityAddress = CreateRandomIdentityAddress();
+        var mockIDeletionProcessLogger = A.Fake<IDeletionProcessLogger>();
+
+        // Act
+        await deleter.Delete(identityAddress, mockIDeletionProcessLogger);
+
+        // Assert
+        A.CallTo(() => mockIDeletionProcessLogger.LogDeletion(identityAddress, AggregateType.Challenges)).MustHaveHappenedOnceExactly();
```

Comment:
Note to myself: should we pass a string instead of an enum? This would improve reusability. But it would require some work in order to be able to pass additional data to an audit log entry.